iterator.go

```diff
@@ -228,8 +228,15 @@ func (iter *Iterator) Next() {
 	}
 
 	node, err := iter.t.next()
-	// TODO: double-check if this error is correctly handled.
-	if node == nil || err != nil {
+	// If an error occurred or no more nodes, update iterator state accordingly
+	if err != nil {
+		iter.err = err
+		iter.t = nil
+		iter.valid = false
+		return
+	}
+
+	if node == nil {
```

Comment:
```suggestion
	// TODO: double-check if this error is correctly handled.
	if node == nil || err != nil {
	    if err != nil {
	        iter.err = err
	    }
```

The change is equivalent to this.

Reply:
thanks for taking the suggestion, but it seems messed up with the indent, 4 spaces vs 1 tab.